administrator/components/com_actionlogs/src/Plugin/ActionLogPlugin.php

```diff
@@ -63,7 +67,8 @@ abstract class ActionLogPlugin extends CMSPlugin
      */
     protected function addLog($messages, $messageLanguageKey, $context, $userId = null)
     {
-        $user = $this->app->getIdentity();
+        $app  = $this->getApplication();
```

Comment:
```suggestion
        $app  = $this->getApplication() ?: $this->app;
```
Here is a fallback needed, when getApplication returns null, to use $this->app, otherwise you force all plugins, which do extend from this class, to inject the application, especially when they use the legacy architecture, then we have a problem. This would be a rather hard BC break.